cli/src/commands/alpha/take_more_labels.rs

```diff
@@ -0,0 +1,260 @@
+///! Task: transfer some labels from one dataset to another:
+///!
+///! For all comments in the target dataset with `only_for_target_label` assigned,
+///! - copy all labels from the source dataset
+///! - nest them under a new target dataset parent label: `transferred_label_parent`
+///! - EXCEPT if the source label starts with the new parent label, don't nest it
+///!
+use std::collections::HashMap;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use colored::Colorize;
+use failchain::ResultExt;
+use reinfer_client::{
+    resources::comment::{Labelling, NewLabelling, Uid},
+    AnnotatedComment, Client, Comment, DatasetIdentifier, Label, LabelName, Sentiment,
+};
+use structopt::StructOpt;
+
+use crate::{
+    errors::{ErrorKind, Result},
+    progress::{Options as ProgressOptions, Progress},
+};
+
+#[derive(Debug, StructOpt)]
+pub struct TakeMoreLabelsArgs {
+    #[structopt(long = "source")]
+    /// Dataset to take labels from.
+    source_dataset: DatasetIdentifier,
+
+    #[structopt(long = "target")]
+    /// Dataset to merge labels into for reviewed comments.
+    target_dataset: DatasetIdentifier,
+
+    #[structopt(long = "only-for-target-label")]
+    /// Only take more labels for target comments with this label.
+    only_for_target_label: LabelName,
+
+    #[structopt(long = "transferred-label-parent")]
+    /// Prefix transferred labels with this parent label.
+    transferred_label_parent: LabelName,
+}
+
+pub fn run(args: &TakeMoreLabelsArgs, client: &Client) -> Result<()> {
+    let TakeMoreLabelsArgs {
+        source_dataset,
+        target_dataset,
+        only_for_target_label,
+        transferred_label_parent,
+    } = args;
+    let source_dataset = client
+        .get_dataset(source_dataset.clone())
+        .chain_err(|| ErrorKind::Client("Getting source dataset failed.".into()))?;
+    let target_dataset = client
+        .get_dataset(target_dataset.clone())
+        .chain_err(|| ErrorKind::Client("Getting target dataset failed.".into()))?;
+
+    let statistics = Arc::new(Statistics::default());
+    let make_progress = || -> Result<Progress> { Ok(get_take_more_labels_progress(&statistics)) };
+    make_progress()?;
+
+    for source_id in target_dataset.source_ids.iter() {
```

Comment:
Does this operation potentially only need to be performed for sources which are included in both datasets?